docs/argocd_app_plugin.md

```diff
@@ -0,0 +1,54 @@
+## 1 ArgoCD Application Plugin
+
+This plugin installs an ArgoCD application.
+
+**Notes:**
+- ArgoCD itself must have been already installed before the usage of this plugin. To install ArgoCD, use [this plugin](https://github.com/merico-dev/stream/blob/main/docs/argocd_plugin.md).
+- Currently, only the Helm chart is supported when creating the ArgoCD application.
+- At the moment, DevStream doesn't support dependency or concurrency yet. So, in the config file, the ArgoCD app plugin must be placed _after_ the ArgoCD plugin if you want to install ArgoCD first then create the ArgoCD application.
+
+## 2 Usage:
+
+```yaml
+tools:
+- name: argocd
+  version: 0.0.1
+  options:
+    repo:
+      name: argo
+      url: https://argoproj.github.io/argo-helm
+    chart:
+      name: argo/argo-cd
+      release_name: argocd
+      namespace: argocd
+      create_namespace: False
+# name of the plugin
+- name: argocdapp
+  # version of the plugin
+  version: 0.0.1
+  # options for the plugin
+  # Checkout the version from the GitHub releases.
```

Comment:
We could change "Checkout" to lowercase "checkout" to keep it consistent with the previous comment.

Reply:
#51